lib/plugins/aws/package/compile/functions.js

```diff
@@ -729,17 +733,20 @@ async function addFileContentsToHashes(filePath, hashes) {
   });
 }
 
-function extractLayerConfigurationsFromFunction(functionProperties, cfTemplate) {
+function extractLayerConfigurationsFromFunction(serverless, functionProperties, cfTemplate) {
   const layerConfigurations = [];
   if (!functionProperties.Layers) return layerConfigurations;
   functionProperties.Layers.forEach((potentialLocalLayerObject) => {
     if (potentialLocalLayerObject.Ref) {
       const configuration = cfTemplate.Resources[potentialLocalLayerObject.Ref];
+
       if (!configuration) {
-        throw new ServerlessError(
-          `Could not find reference to layer: ${potentialLocalLayerObject.Ref}. Ensure that you are referencing layer defined in your service.`,
-          'LAMBDA_LAYER_REFERENCE_NOT_FOUND'
-        );
+        if (process.env.SLS_DEBUG) {
+          serverless.cli.log(
```

Comment:
It would be better to rely on `@serverless/utils/log`

We're rather on the path on removing dependency on `serverless` instance.

Reply:
of course, great suggestion :+1: